flask/ctx.py

```diff
@@ -133,20 +135,25 @@ def copy_current_request_context(f):
         def index():
             @copy_current_request_context
             def do_some_work():
-                # do some work here, it can access flask.request like you
-                # would otherwise in the view function.
+                # do some work here, it can access flask.request or
+                # flask.session like you would otherwise in the view function.
                 ...
             gevent.spawn(do_some_work)
             return 'Regular response'
 
     .. versionadded:: 0.10
+
+    .. versionchanged:: 1.0.3
+       A copy of the current session object is added to the request context
+       copy. This prevents `flask.session` pointing to an out-of-date object.
     """
     top = _request_ctx_stack.top
     if top is None:
         raise RuntimeError('This decorator can only be used at local scopes '
             'when a request context is on the stack.  For instance within '
             'view functions.')
     reqctx = top.copy()
+    reqctx.session = deepcopy(session)
```

Comment:
This should be added to `RequestContext.copy()` instead, along with the docs changes.